packages/adapter/addon/rest.js

```diff
@@ -1301,9 +1301,11 @@ function fetchSuccessHandler(adapter, payload, response, requestData) {
 
 function fetchErrorHandler(adapter, payload, response, errorThrown, requestData) {
   let responseData = fetchResponseData(response);
+  payload = adapter.parseErrorResponse(payload);
+
   if (responseData.status === 200 && payload instanceof Error) {
     responseData.errorThrown = payload;
-    payload = responseData.errorThrown.payload;
+    payload = responseData.errorThrown;
```

Comment:
Do you think we need this reassignment to `payload`?

Reply:
Indeed, that was an error on my side. 
I shouldn't even change anything in that branch. 
My fix was to handle the case where payload is a javascript string representing a JSONAPI error object. In that branch, payload is an Error object, so, I shouldn't change anything here.